### PR TITLE
Added blog post and table of patches.

### PR DESCRIPTION
**Pull request by jberkus**
This contains Dan's post on Docker patches, as well as a visually organized table of patches we maintain.

@rhatdan, questions:

1. Your blog post mentions backporting fixes, but none of the listed patches are of that type.
2. Is there a PR or Issue for Improved-searching-experience?
3. If you can build the site, please check out the new docker_patches doc I created to see if that works for you.

@bproffitt, I took one pass at editing this, but it's a LOT of text.  Please look for basic spelling/grammar issues I might have missed?

### CODE REVIEW

**Comment by bproffitt** [comment]
Files edited. When ready, this can be posted (at least, from an editorial point of view).

**Comment by rhatdan** [comment]
I am sure the formatting is fine.

**Comment by bproffitt** [comment]
It looked good to me when I previewed it in Atom

File: source/docs/docker_patches.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+* [11991](https://github.com/docker/docker/pull/11991)
+* [10411](https://github.com/docker/docker/pull/10411)
+
+**Status**: Upstream docker does not like this patch set, because they want the **docker** experience to be the same everywhere.  If I do a `docker pull fedora`, I always get it from docker.io.  We and our customers do not agree, so we continue to carry this patch.  We prefer the yum/apt method for specifying registries.
+
+### Improved-searching-experience.patch
+
+Red Hat wants to allow users to search multiple registries as described above. This patch improves the search experience.
+
+**Status**:  This patch actually works with the Add-add-registry-and-block-registry-options-to-docke.patch so we need to carry it also.
```
Comment (nzwulfin) [inline source/docs/docker_patches.md]: Add-add-registry-and-block-registry-options-to-docke.patch <-- docker.patch

Comment (jberkus) [inline source/docs/docker_patches.md]: hmmm, I thought I fixed that.

Comment (jberkus) [inline source/docs/docker_patches.md]: ah, I fixed the first one, but not the second.  Thanks!

**Comment by jberkus** [comment]
@rhatdan for the docker_patches doc, could we get a list of backport patches?

**Comment by rhatdan** [comment]
@jberkus that varies depending on the version of docker you are using.  IE Docker-1.10 has more back ported patches then docker-1.12.

Here is the current patches for docker-1.12

https://github.com/docker/docker/compare/v1.12.0...projectatomic:docker-1.12

Here are patches for docker-1.10.3

https://github.com/docker/docker/compare/v1.12.0...projectatomic:rhel7-1.10.3

**Comment by jberkus** [comment]
@rhatdan that's *all* the commits, though. It's not a useful view of anything. 

I'd still like to add a table of backports to the docker_patches page.  It would be valuable to our users, as well as making our patching work more attractive to 3rd parties.  What I'd see is a table like this:

| Patch                  | From Ver | To Versions | Status |
| -----------------------|---------------|-----------------|-----------|
| dns-resolution-fix-2 | 1.11 | 1.10, 1.9 | current |
| add-dir-chmod | 1.10 | 1.9 | archive |

We don't have to do it immediately, but I'd like to have someone working on it.

**Comment by rhatdan** [comment]
Sort of a living document then, which would be updated each time we add a patch to an older version of docker.  Would be nice if we had a tool to do this or at least grab the data.

**Comment by rhatdan** [comment]
@runcom What do you think?

**Comment by runcom** [comment]
@rhatdan LGTM :)

@rhatdan @jberkus I can work on a simple tool to format out a table with backports and stuff from there - shouldn't be that hard

**Comment by jberkus** [comment]
@rhatdan yes, exactly.  BTW, I'd also like the table on that page to be a "living document", in that it should get updated every time you add a new patch or change status for an old one.  If you don't feel like editing it, file an issue when that happens.

**Comment by jberkus** [comment]
I'll push this tommorrow morning if there's no blockers.

**Pull request merged**